contracts/YakRegistry.sol

```diff
@@ -1,40 +1,76 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
 
 import "./lib/Ownable.sol";
+import "./lib/EnumerableSet.sol";
+import "./YakStrategy.sol";
 
 /**
  * @notice YakRegistry is a list of officially supported strategies.
  * @dev DRAFT
  */
 contract YakRegistry is Ownable {
+    uint256 public strategiesLength;
+    mapping(address => uint256[]) public strategyIdsForDepositToken;
 
-    mapping(address => bool) public registeredStrategies;
+    mapping(address => bool) private registeredStrategies;
+    address[] private strategies;
```

Comment:
Consider using Enumerable Set, it has methods like Has that checks if a certain value belongs to the Enumerable Set, which is somewhat the things you're implementing yourself here

Reply:
I did consider Enumerable Set, but decided it doesn't fit right. We could only replace the registeredStrategies mapping, since they don't guarantee not to reorder on add/remove and we use the index as id and don't want that.
If we replace only registeredStrategies the only benefit would be that we don't need to create our own "has" functionality. The downside is we would in reality use an array instead of mapping, but mapping is more gas efficient.